frontend/packages/operator-lifecycle-manager/src/components/create-catalog-source.tsx

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { ActionGroup, Button, Form, FormGroup, TextInput } from '@patternfly/react-core';
+import {
+  ButtonBar,
+  history,
+  NsDropdown,
+  withHandlePromise,
+  HandlePromiseProps,
+} from '@console/internal/components/utils';
+import { RadioGroup } from '@console/internal/components/radio';
+import { k8sCreate } from '@console/internal/module/k8s';
+import { CatalogSourceModel } from '../models';
+
+enum AvailabilityValues {
+  ALL_NAMESPACES = '0',
+  SINGLE_NAMESPACE = '1',
+}
+
+const availabilityKinds = [
+  {
+    value: AvailabilityValues.ALL_NAMESPACES,
+    title: 'All namespaces on the cluster',
+    desc: 'Catalog will be available in all namespaces',
+  },
+  {
+    value: AvailabilityValues.SINGLE_NAMESPACE,
+    title: 'A specific namespace on the cluster',
+    desc: 'Catalog will only be available in a single namespace',
+  },
+];
+
+export const CreateCatalogSource: React.FC<HandlePromiseProps> = withHandlePromise(
+  ({ handlePromise, inProgress, errorMessage }) => {
+    const [availability, setAvailability] = React.useState(AvailabilityValues.ALL_NAMESPACES);
+    const [image, setImage] = React.useState('');
+    const [displayName, setDisplayName] = React.useState('');
+    const [name, setName] = React.useState('');
+    const [namespace, setNamespace] = React.useState();
```

Comment:
Is this a string or object? If a string, consider `React.useState<string>()` or `React.useState('')`